script/upload

```diff
@@ -0,0 +1,262 @@
+#!/bin/sh
+
+set -e
+
+REPO="git-lfs/git-lfs"
+WORKDIR="$(mktemp -d)"
+
+trap 'rm -fr "$WORKDIR"' EXIT
+
+say () {
+  [ -n "$QUIET" ] && return
+  local format="$1"
+  shift
+  printf "$format\n" "$@" >&2
+}
+
+abort () {
+  local format="$1"
+  shift
+  printf "$format\n" "$@" >&2
+  exit 2
+}
+
+curl () {
+  command curl -nfSs "$@"
+}
+
+categorize_os () {
+  local os="$1"
+
+  if [ "$os" = "freebsd" ]
+  then
+    echo FreeBSD
+  else
+    ruby -e 'puts ARGV[0].capitalize' "$os"
+  fi
+}
+
+categorize_arch () {
+  local arch="$1"
+
+  ruby -e 'puts ARGV[0].upcase' "$arch"
```

Comment:
Hmm... I'm hesitant to introduce new Ruby dependencies in the Git LFS code base, since I think that it would be nice to remove it (in particular people who build Git LFS from source would like this, since we can build the documentation without Ronn).

I'm not a Ruby expert by any means (I'm barely familiar with it), but I think that this line could instead be released with:

```bash
echo "$arch" | tr -s a-z A-Z
```

Reply:
This portion could probably be replaced. However, I should point out that this script is only designed to be run by a maintainer. End users will never want to run this script and any attempt to do so will fail due to lack of permissions. I think it's safe to assume that a reasonably modern Ruby is available on macOS and Linux systems, which are the systems on which Git LFS maintainers are likely to be doing releases.

As you've pointed out below, there are some tricky places where we really would like to produce something that shell scripting can't provide us natively. However, since we require `prove`, we can replace the Ruby with Perl instead, if you like. I felt Ruby would be more familiar to current and future maintainers and it wouldn't be a problem to use it since we already use `ronn`, but if you'd prefer Perl, I can go along with that.

Reply:
Hm, I think that you make an excellent point about this script being run by a maintainer, and hence its contents do not add to our dependency list. Given this, I think that the Ruby is fine to stick around without issue.